### PR TITLE
Use KV cache till input seq len for prefill phase

### PR DESCRIPTION
**Pull request by puneeshkhanna**
Pad KV cache to full input + new tokens len for decode phase. Delete the KV cache used as inputs by HPU graphs after full prompt generation. Ensure KV cache is not returned as output tensor during decode phase. Deletion of KV cache input tensor used by HPU graphs needs to be protected by PT_HPUGRAPH_DISABLE_TENSOR_CACHE env variable.
All the changes are protected by bucket internal flag.

# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by puneeshkhanna** [comment]
Updated command (remove --reuse_cache and prepend PT_HPUGRAPH_DISABLE_TENSOR_CACHE=1)

PT_HPUGRAPH_DISABLE_TENSOR_CACHE=1 python ../gaudi_spawn.py --use_deepspeed --world_size 8 run_generation.py --model_name_or_path /mnt/weka/data/pytorch/llama2/Llama-2-70b-hf/ --use_hpu_graphs --use_kv_cache --max_input_tokens 2048 --max_new_tokens 2048 --batch_size 200 --attn_softmax_bf16  --trim_logits --bf16  --warmup 2 --n_iterations 2 --limit_hpu_graphs --bucket_internal --bucket_size 128

Also requires pytorch-integration patch - https://gerrit.habana-labs.com/#/c/408363/

File: optimum/habana/transformers/models/llama/modeling_llama.py [hunk 1]
```diff
@@ -300,9 +299,8 @@ def pre_attn_forward(
                 past_key_value = (self.k_cache.get_shape(), self.v_cache.get_shape())
             else:
                 if past_key_value is None:
-                    past_key = torch.zeros(key_states.shape, dtype=self.k_proj.weight.dtype, device=key_states.device)
-                    past_value = torch.zeros(key_states.shape, dtype=self.k_proj.weight.dtype, device=key_states.device)
-                    past_key_value = (past_key, past_value)
+                    # Return list instead of tuple
```
Comment (bgoldberg-habana) [inline optimum/habana/transformers/models/llama/modeling_llama.py:304]: this won't work for fp8 flow, need to revert, and maybe add a comment why it was added at the first place (allocating the kv cache according to k_proj type which can be bf16/fp8

Comment (puneeshkhanna) [reply optimum/habana/transformers/models/llama/modeling_llama.py:304]: @bgoldberg-habana  - Fixed.

**Review by dvarshney-habana — APPROVED** [review]

**Comment by astachowiczhabana** [comment]
https://github.com/huggingface/optimum-habana/pull/1028
